### PR TITLE
test(e2e): adding WebdriverIO e2e test structure

### PR DESCRIPTION
**Pull request by jeffchew**
### Related Ticket(s)

#2984

### Description

This adds the core WebDriverIO test harness, Github action, and first initial tests that were converted from the Python script versions. The strategy taken is to use direct urls for the full screen view of the canvas. This includes passing in query params if necessary, rather than adjusting knobs.

There are still remaining .py files, which need to be converted to e2e.js files and in WebdriverIO format. I think it will be ok to tackle this in separate PRs.

### Changelog

**New**

- WebdriverIO e2e testing architecture
- e2e tests for `CalloutQuote`, `ContentBlockMixed`, `CardSectionImages`, `CardSectionSimple`

### CODE REVIEW

**Comment by jeffchew** [comment]
One thing I might need to change, the github actions are currently targeting the canary environment. I'll need to hook up the harness for this to run on the PR level and isolated deploy preview, which will probably need to happen on jenkins. This action can otherwise run on master whenever new changes happen, which can continue to happen on GH actions.

**Comment by asudoh** [comment]
Looks a progress @jeffchew, the only feedback I have is striving to capture events where the UI becomes in the state we are testing against, instead of running wait calls that assumes the UI becomes such state in certain seconds. Such approach makes the test less flaky; Flakiness is the biggest thing that drives people away from E2E testing.

**Comment by jeffchew** [comment]
> Looks a good progress @jeffchew, the only feedback I have is it's good if we strive to capture events where the UI becomes in the state we are testing against, instead of running wait calls that assumes the UI becomes such state in certain seconds. Such approach makes the test faster and less flaky; Slowness and flakiness is the biggest thing that drives people away from E2E testing.

Good call, any thoughts on modifying the tests to check for that? We could also possibly do this in another PR too, and just get the initial architecture in place first.

File: packages/react/src/patterns/blocks/CalloutQuote/__tests__/CalloutQuote.e2e.js [hunk 1]
```diff
@@ -0,0 +1,113 @@
...
+ * @private
+ */
+const _path = '/iframe.html?id=patterns-blocks-calloutquote--default';
+
+describe('CalloutQuote', () => {
+  beforeAll(() => {
+    browser.url(_url + _path);
+    browser.setWindowSize(1200, 800);
+    browser.pause(1000);
+  });
```
Comment (kennylam) [inline packages/react/src/patterns/blocks/CalloutQuote/__tests__/CalloutQuote.e2e.js:30]: @jeffchew @asudoh Regarding testing UI, perhaps something like this? 

```suggestion
  beforeAll(() => {
    document.addEventListener("DOMContentLoaded", function(event) {
      browser.url(_url + _path);
      browser.setWindowSize(1200, 800);
      browser.pause(1000);
      done();
    });
  });
```

Comment (asudoh) [reply packages/react/src/patterns/blocks/CalloutQuote/__tests__/CalloutQuote.e2e.js:30]: @kennylam Exactly, the notion is tracking for events that lets us proceed to the next step. It'll depend on each wait calls seen in this PR. And agreed with @jeffchew that we can do it in separate PRs.

**Pull request merged**